src/extension.js

```diff
@@ -55,25 +58,216 @@ module.exports.activate = function( context ) {
+			console.log("Test result JSON", { json });
+
+
+
+			// Caught errored
+			if (json.totalError > 0) {
+				run.errored(test, "Not implemented");
+			}
+			// Caught test failures 
+			if (json.totalFail > 0) {
+				run.failed(test, "Test failed");
+			}
+			else {
+				run.passed(test);
+			}
+		} catch (error) {
+			// Log and mark the test as failed
+			LOG.error(`Error running test ${test.label}: ${error}`);
+			// run.failed(test, error.message);
+			run.errored(test, error.message);
+		}
+	}
+	run.end();
+}
+
+
+/**
+ * Applies path mappings to a given relative file path.
+ *
+ * This function retrieves path mappings from the VSCode configuration for "testbox"
+ * and replaces the start of the relative file path with the corresponding target path
+ * if a matching source path is found.
+ *
+ * @param {string} relativeFilePath - The relative file path to apply mappings to.
+ * @returns {string} - The new file path after applying the mappings, or the original
+ *                     relative file path if no mappings match.
+ */
+function applyPathMappings(relativeFilePath) {
+	const mappings = vscode.workspace.getConfiguration("testbox").get("pathMappings");
+
+	for (const mapping of mappings) {
+		if (relativeFilePath.startsWith(mapping.source)) {
+			// This might not be correct, just have to replace the start
+
+			const newPath = mapping.target + relativeFilePath.slice(mapping.source.length);
+			return newPath;
+		}
+	}
+
+	return relativeFilePath;
+}
+
+/**
+ * Discovers test files and adds them to the test controller.
+ *
+ * This function clears out any previous test items from the controller,
+ * retrieves the list of test files based on the specified glob pattern
+ * and excluded paths from the workspace configuration, and then maps
+ * each file path to a test name before adding it to the controller.
+ *
+ * @param {vscode.TestController} controller - The test controller to which discovered tests will be added.
+ * @returns {Promise<void>} A promise that resolves when the test discovery is complete.
+ */
+async function discoverTests(controller) {
+	// Clear out previous items
+	let runnerUrl = vscode.workspace.getConfiguration("testbox").get("runnerUrl");
+	// console.log(runnerUrl);
+	controller.items.replace([]);
+
+
+	const excludedPaths = vscode.workspace.getConfiguration("testbox").get("excludedPaths");
+	const files = await vscode.workspace.findFiles(testFileGlob, excludedPaths);
+	for (const file of files) {
+		const mappedPath = applyPathMappings(vscode.workspace.asRelativePath(file.fsPath));
+		const testName = convertTestPathToTestName(mappedPath);
+		const testUrl = runnerUrl + "?reporter=JSON&recurse=true&testBundles=" + testName;
+		// If we have a suite we should also add &testSuites=testsuite
+		// const testItem = controller.createTestItem(file.fsPath, testName, file);
+		const testItem = controller.createTestItem(testUrl, testName, file);
+		// testItem.canResolveChildren = true;
```

Comment:
I will add this later, this would show all the describe and it type specs, but it requires a bit more parsing (as they are fun to parse!